larray/core.py

```diff
@@ -4685,6 +4726,144 @@ def rename(self, renames=None, to=None, inplace=False, **kwargs):
+        """Reorder and/or add new labels in axes.
+
+        Place NaN or given `fill_value` in locations having no value previously.
+
+        Parameters
+        ----------
+        axes_to_reindex : axis ref or dict {axis ref: axis} or list of tuple (axis ref, axis)
+                          or list of Axis or AxisCollection
+            Axes to reindex. If a single axis reference is given, the `new_axis` argument must be provided.
+            If a list of Axis or an AxisCollection is given, all axes will be reindexed by the new ones.
+            In that case, the number of new axes must match the number of the old ones.
+        new_axis : , optional
+            New axis if `axes_to_replace` contains a single axis reference.
+        fill_value : scalar, optional
+            Value set to data corresponding to added labels.
+            By defaults NaN.
+        inplace : bool, optional
+            Whether or not to modify the original object or return a new array and leave the original intact.
+            By defaults False.
+        **kwargs : Axis
+            New axis for each axis to reindex given as a keyword argument.
+
+        Returns
+        -------
+        LArray
+            Array with reindexed axes.
+
+        Notes
+        -----
+        When introducing NAs into an array containing integers via reindex_axis,
+        all data will be promoted to float in order to store the NAs.
+
+        Examples
+        --------
+        >>> arr = ndtest((2, 2, 2))
+        >>> arr
+         a | b\\c | c0 | c1
+        a0 |  b0 |  0 |  1
+        a0 |  b1 |  2 |  3
+        a1 |  b0 |  4 |  5
+        a1 |  b1 |  6 |  7
+        >>> arr2 = ndtest((2, 3, 3))
+        >>> arr.reindex(arr2.axes)
+         a | b\\c |  c0 |  c1 |  c2
+        a0 |  b0 | 0.0 | 1.0 | nan
+        a0 |  b1 | 2.0 | 3.0 | nan
+        a0 |  b2 | nan | nan | nan
+        a1 |  b0 | 4.0 | 5.0 | nan
+        a1 |  b1 | 6.0 | 7.0 | nan
+        a1 |  b2 | nan | nan | nan
+        >>> b = Axis('b', ['b1', 'b2', 'b0'])
+        >>> c = Axis('c', ['c2', 'c1', 'c0'])
+        >>> arr.reindex({'b' : b, 'c' : c}, fill_value=-1)
+         a | b\\c | c2 | c1 | c0
+        a0 |  b1 | -1 |  3 |  2
+        a0 |  b2 | -1 | -1 | -1
+        a0 |  b0 | -1 |  1 |  0
+        a1 |  b1 | -1 |  7 |  6
+        a1 |  b2 | -1 | -1 | -1
+        a1 |  b0 | -1 |  5 |  4
+        >>> arr.reindex({x.b : b, x.c : c})
+         a | b\\c |  c2 |  c1 |  c0
+        a0 |  b1 | nan | 3.0 | 2.0
+        a0 |  b2 | nan | nan | nan
+        a0 |  b0 | nan | 1.0 | 0.0
+        a1 |  b1 | nan | 7.0 | 6.0
+        a1 |  b2 | nan | nan | nan
+        a1 |  b0 | nan | 5.0 | 4.0
+        """
+        if not np.isscalar(fill_value):
+            raise TypeError("fill_value must be a scalar")
+        res_axes = self.axes.replace(axes_to_reindex, new_axis, **kwargs)
+        res = full(res_axes, fill_value, dtype=common_type([self.data, [fill_value]]))
```

Comment:
you can use common_type((self.data, fill_value)) instead (ie no [] necessary around fill_value)